### PR TITLE
Add support for Terraform v0.9

### PR DESCRIPTION
**Pull request by ncs-sburns**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ kitchen-terraform is a set of [Test Kitchen] plugins for testing
 
 - [Bundler] **(~> 1.12)**
 
-- [Terraform] **(>= 0.6, < 0.9)**
```
Comment (aaron-lane) [inline README.md]: Support for Terraform 0.6 should not be removed until kitchen-terraform 1.0.

File: kitchen-terraform.gemspec [hunk 2]
```diff
@@ -19,7 +19,8 @@ require 'terraform/project_version.rb'
     'Michael Glenney',
     'Nell Shamrell-Harrington',
     'Nick Willever',
-    'Walter Dolce'
+    'Walter Dolce',
+    'Steven A. Burns'
```
Comment (aaron-lane) [inline kitchen-terraform.gemspec]: This list is being maintained in alphabetical order.

File: lib/terraform/version.rb [hunk 3]
```diff
@@ -31,15 +31,15 @@ def self.create(value:)
     end
 
     def self.deprecated
-      [new(value: '0.6')]
+      [new(value: '0.7')]
```
Comment (aaron-lane) [inline lib/terraform/version.rb]: Terraform 0.7 is not yet deprecated.

**Comment by ncs-alane** [comment]
Whoops, commented from my personal account. 🙀

File: kitchen-terraform.gemspec [hunk 4]
```diff
@@ -79,7 +80,7 @@ require 'terraform/project_version.rb'
 
   specification.required_ruby_version = ['>= 2.1', '< 2.5']
 
-  specification.requirements = ['Terraform >= 0.6, < 0.9']
+  specification.requirements = ['Terraform >= 0.7, < 0.10']
```
Comment (ncs-alane) [inline kitchen-terraform.gemspec]: Terraform 0.6 is still supported.

File: lib/terraform/version.rb [hunk 5]
```diff
@@ -35,11 +35,11 @@ def self.deprecated
     end
 
     def self.latest
-      new value: '0.8'
+      new value: '0.9'
     end
 
     def self.supported
-      [latest, new(value: '0.7'), *deprecated]
+      [latest, new(value: '0.8'), *deprecated]
```
Comment (ncs-alane) [inline lib/terraform/version.rb]: 0.7 should also be in this list.

**Comment by ncs-alane** [comment]
This may be blocked by https://github.com/hashicorp/terraform/issues/12969; waiting for analysis.

**Comment by ncs-alane** [comment]
Fixes #112 

Are there any changes to the workflow necessary to accommodate backends and remote state?

**Comment by RyanJarv** [comment]
Getting the following with this branch with 0.9.2:

```Failed to load backend: State path cannot be specified with a plan file. The plan itself contains the state to use. If you wish to change that, please create a new plan and specify the state path when creating the plan.```

Simply removing the state flag file from the apply command works.

**Comment by aaron-lane** [comment]
@ryanjarv:

Thank you for your interest in the project!

Removing the state flag lead to the discovery of a bug which has been reported as hashicorp/terraform#12969. Once that bug is solved, we will be able to finish this feature and release a new version.

**Comment by RyanJarv** [comment]
Hm working for me currently, I think that was resolved in 0.9.1 or 0.9.2 maybe?

**Comment by aaron-lane** [comment]
@ryanjarv:

I may have misunderstood your comment. Are you saying that 0.9.2 properly saves the state file path in the plan?

**Comment by RyanJarv** [comment]
I believe so, will do some more testing and let you know if I run into anything.

**Comment by RyanJarv** [comment]
One other thing that changed is `terraform init` needs to be run to set up the backend and also does the job of `terraform get`. I'm just running that before hand right now but I think that makes sense to have kitchen do that though.

**Comment by RyanJarv** [comment]
Waiting on a kitchen test to finish, but converge/test works for me currently with https://github.com/RyanJarv/kitchen-terraform. It's just hacked together but figured I'd link to it for reference.

**Comment by ncs-alane** [comment]
@ncs-sburns: do you have time to move this branch along?

**Comment by LosLocosTacos** [comment]
Still fails with Terraform 0.9.2 unless I use @RyanJarv 's workarounds.

**Comment by ncs-sburns** [comment]
Still fails with Terraform 0.9.2 unless I use @RyanJarv 's workarounds.

**Comment by RyanJarv** [comment]
Does anyone know what the behavior should be on different versions? Seems the state file is just removed in every case on 0.9+, I think the issue then is how to use different test suites on the same repo, I haven't gotten into that much so not sure how that is currently handled. If it's done with different state files before wondering if it makes sense to use the new env's feature for that on 0.9+.

**Comment by RyanJarv** [comment]
I should mention that I'm pretty much just thinking in the context of backends on 0.9+. Not sure how the state file is handled locally, or if backends was something that was even supported before (would have had to run `terraform remote` on the repo before and realize I haven't seen that in the repo yet). Not sure if it's the same issue locally or not, but if 0.9 works locally now I don't see backends as a blocker for getting support for that out.

**Comment by ncs-alane** [comment]
@RyanJarv: https://github.com/hashicorp/terraform/issues/12969 is the blocker for adding support for Terraform 0.9.

**Comment by RyanJarv** [comment]
Ah ok, weird.. thought I had the same issue in 9.0 resolved in 9.1, but maybe not. Makes sense I wasn't running into it here since I'm only using remote state. Now that I think about it I'm kinda wondering what the behavior should be when remote state is included in the config or if terraform will let you override that.

**Comment by RyanJarv** [comment]
Might be worth mentioning terraform 0.9 has support for different environments which I believe is similar to managing different state files.

**Comment by ncs-alane** [comment]
@RyanJarv: in the interest of expedience, I want to focus on integrating Terraform 0.9 in to the existing workflow for kitchen-terraform 0.7 and then look at leveraging new features of Terraform in kitchen-terraform 0.8.

**Pull request merged**